packages/run-protocol/src/reserve/assetReserve.js

```diff
@@ -77,20 +90,74 @@ const start = async (zcf, privateArgs) => {
    */
   const addIssuer = async (issuer, keyword) => {
     const brand = await E(issuer).getBrand();
+    trace('addIssuer', { brand, keyword });
     assert(
       keyword !== RunKW && brand !== runKit.brand,
       `${RunKW} is a special case handled by the reserve contract`,
     );
 
-    const liquidityIssuer = E(ammPublicFacet).getLiquidityIssuer(brand);
+    trace('addIssuer storing', {
+      keyword,
+      brand,
+    });
+
+    saveBrandKeyword(brand, keyword);
+    await zcf.saveIssuer(issuer, keyword);
+  };
+
+  /**
+   * @param {Issuer} baseIssuer on which the liquidity issuer is based
+   */
+  const addLiquidityIssuer = async baseIssuer => {
+    const getBrand = () => {
+      try {
+        return zcf.getBrandForIssuer(baseIssuer);
+      } catch {
+        assert.fail(
+          `baseIssuer ${baseIssuer} not known; try addIssuer() first`,
+        );
+      }
+    };
+    const baseBrand = getBrand();
+    const baseKeyword = keywordForBrand.get(baseBrand);
+    const liquidityIssuer = E(ammPublicFacet).getLiquidityIssuer(baseBrand);
     const liquidityBrand = await E(liquidityIssuer).getBrand();
+    const liquidityKeyword = makeLiquidityKeyword(baseKeyword);
 
-    brandForKeyword.init(keyword, [brand, liquidityBrand]);
-    keywordForBrand.init(brand, keyword);
-    return Promise.all([
-      zcf.saveIssuer(issuer, keyword),
-      zcf.saveIssuer(liquidityIssuer, makeLiquidityKeyword(keyword)),
-    ]);
+    trace('addLiquidityIssuer', {
+      baseBrand,
+      baseKeyword,
+      liquidityBrand,
+      liquidityKeyword,
+    });
+    saveBrandKeyword(liquidityBrand, liquidityKeyword);
+    liquidityBrandForBrand.init(baseBrand, liquidityBrand);
+
+    await zcf.saveIssuer(liquidityIssuer, liquidityKeyword);
+  };
+
+  let keywordNum = 0;
+  /**
+   * @param {Brand} secondaryBrand
+   */
+  const addIssuerFromAmm = async secondaryBrand => {
+    assert(
+      secondaryBrand !== runKit.brand,
+      `${RunKW} is a special case handled by the reserve contract`,
+    );
```

Comment:
>If addIssuerFromAMM is public, it has to verify that the brand is a secondary brand (not a liquidity brand) with the AMM before adding it. Otherwise, someone could bollix up our careful plans by calling addIssuerFromAmm(doubloonLiquidityBrand).

Had to read this a few times. Repeating what I got:
- this design has `addIssuerFromAMM` public which means it has to verify the brand argument is a secondary brand with the AMM before adding it
- the line `const issuer = await E(ammPublicFacet).getIssuer(ammSecondaryBrand);` can return a liquidity brand
- therefore someone could bollix up our careful plans by calling addIssuerFromAmm(doubloonLiquidityBrand).

> Why doesn't it work to add both issuers when someone (the AMM?) calls addIssuerFromAmm?

IIRC, the liquidity issuer wasn't yet defined on the AMM so the `E(ammPublicFacet).getLiquidityIssuer` call would fail. 

Let's chat about whether we can simplify this.